snapcraft/plugins/dotnet.py

```diff
@@ -140,3 +140,9 @@ def _get_appname(self):
             if fnmatch.fnmatch(file, '*.??proj'):
                 return os.path.splitext(file)[0]
                 break
+
+    def env(self, root):
+        if root == self.installdir:
```

Comment:
This logic is _slightly_ unusual. Would you mind adding a comment specifying why you only want this done for the installdir?